docs/manual/src/renaming.md

```diff
@@ -0,0 +1,40 @@
+# Renaming
+
+UniFFI provides two ways to customize names in generated bindings:
+
+1. **Proc-macro attributes** - Use variants of `#[uniffi(name = "...")]` to rename items directly in Rust code. See [proc-macro renaming](proc_macro/renaming.md).
+2. **TOML configuration** - Define language-specific renames in `uniffi.toml`.
+
+## TOML-based Renaming
+
+You can rename types, functions, methods, constructors, and their members (fields, variants, arguments) on a per-language basis using `uniffi.toml`.
```

Comment:
How does this work with primary constructors, which are often special cased and don't have names?  After skimming the code, I think it might not have any effect but I'm not sure.  Any behavior is fine with me for this corner case, but let's document it.